src/daemon/daemon.cpp

```diff
@@ -493,6 +494,11 @@ try // clang-format on
 
     return grpc::Status::OK;
 }
+catch (const mp::StartException& e)
+{
+    vm_instances.erase(e.name());
```

Comment:
I think this to also do:

config->factory->remove_resources_for(name);
config->vault->remove(name);
vm_instance_specs.erase(name);